packages/@aws-cdk/aws-dynamodb/lib/table-time-to-live-handler/index.ts

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable no-console */
+import type { IsCompleteRequest, IsCompleteResponse, OnEventRequest, OnEventResponse } from '@aws-cdk/custom-resources/lib/provider-framework/types';
+import { DynamoDB } from 'aws-sdk'; // eslint-disable-line import/no-extraneous-dependencies
+import { disableTimeToLive, enableTimeToLive, isTimeToLiveCorrect, isTimeToLiveStable } from './utils';
+
+const dynamodb = new DynamoDB({ apiVersion: '2012-08-10' });
+
+export async function onEventHandler(event: OnEventRequest): Promise<OnEventResponse> {
+  console.log('Event: %j', event);
+
+  if (event.RequestType === 'Create' || event.RequestType === 'Update') {
+    const currentTtl = await dynamodb.describeTimeToLive({
+      TableName: event.ResourceProperties.TableName,
+    }).promise();
+    console.log('Describe time to live: %j', currentTtl);
+
+    switch (currentTtl.TimeToLiveDescription?.TimeToLiveStatus) {
+      case 'ENABLING':
+        throw new Error('Can not change time to live while it is being enabled.');
+        break;
+      case 'DISABLING':
+        throw new Error('Can not change time to live while it is being disabled.');
+        break;
+      case 'ENABLED':
+        if (event.ResourceProperties.TimeToLiveAttribute !== currentTtl.TimeToLiveDescription.AttributeName) {
+          await disableTimeToLive(event);
+        }
+        break;
+      case 'DISABLED':
+        if (event.ResourceProperties.TimeToLiveAttribute) {
+          await enableTimeToLive(event);
+        }
+        break;
+    }
+
+    console.log('Update table: %j', event.ResourceProperties.TableName);
+    return { PhysicalResourceId: event.PhysicalResourceId ?? event.LogicalResourceId };
+  }
+
+  return { PhysicalResourceId: event.PhysicalResourceId };
+}
+
+export async function isCompleteHandler(event: IsCompleteRequest): Promise<IsCompleteResponse> {
+  console.log('Event: %j', event);
+
+  const data = await dynamodb.describeTable({
+    TableName: event.ResourceProperties.TableName,
+  }).promise();
+  console.log('Describe table: %j', data);
```

Comment:
Do we need to describe the table here? I mean: is it possible that `describeTimeToLive()` returns `ENABLED` or `DISABLED` for a table that is not `ACTIVE`?

Reply:
I am not sure if this is possible, but as I took the replica handler as inspiration, I included it to be safe. If you would like we can remove this additional check.

Reply:
To be tested but I would remove it if it's no necessary.

Reply:
I have removed this, as it is probably unnecessary.